### DISCUSSION TITLE
UART Service Encoding

### DISCUSSION THREAD

**Discussion by szabomFB** [ThomasGerstenberg/blatann#80] [Q&A] [answered]
I am trying to better understand the encoding of messages used by central_uart_service example (also quick side note, there is a typo in the title for this, labelling it centEral_uart_service), as I am not receiving any of the expected response from my BLE Nordic UART peripheral device.

I assume the Nordic UART baud rate is set by some standard and the user selectable baud rate refers to the COM port communication with the nRF52840 dongle that I am using. The only other setting I can think of is that in the app I need to specify that I want to use CR+LF to terminate messages- could this be missing in the packets sent via blatann?

I have also tried monitoring the signals on a BLE sniffer, where I can successfully capture the advertising, scanning, and connecting messages, but I do not see any indication of message transfers so I cannot further debug the exact hex bytes sent over BLE...

For reference, here is the full logger output that I see from the connection (I only added a few additional debug info to get dongle's transmitting address and include peripheral name in messages): [BLEConnectLog.txt](https://github.com/ThomasGerstenberg/blatann/files/6518014/BLEConnectLog.txt)
As well as the wireshark sniffer results: [BLEConnectSniff.zip](https://github.com/ThomasGerstenberg/blatann/files/6518033/BLEConnectSniff.zip)

**Comment by szabomFB** [comment, marked as answer]
Added CRLF characters to end of data message (data = data + "\r\n") resolved my issue. I am now seeing a proper response from device

**Reply by ThomasGerstenberg** [reply]
Glad you figured it out. Yeah the UART service itself makes no assumption about the data being sent (raw, ascii, utf8, line endings, etc.) to support any type of data transmitted, so the CRLF needs to be at the application level
